src/ipmanager/api/views.py

```diff
@@ -72,29 +72,34 @@ def get(self, request):
     if 'group' not in request.GET:
       return JsonResponse({"@id":request.build_absolute_uri(), 
                            "ip":ip,
-                           # Demo-only data, with actual data iterate through all group_keys and make url and parse response accordingly
-                           "checks":[{"@id":"https://ipmanager-test.lib.umd.edu/check?group=Henson00001&ip=127.0.0.1",
-                                      "group":{"@id":"https://ipmanager-test.lib.umd.edu/groups/Henson00001",
-                                               "key":"henson20001","name":"Henson Collection"},
-                                      "ip_address":ip,
-                                      "contained":False}]})
+                           "checks":generateChecksArray(ip, all_groups, request)})
     else:
-      group = request.GET['group']
-      if group in group_keys:
+      group_key = request.GET['group']
+      try:
+        db_group = Group.objects.get(key=group_key)
         return JsonResponse({"@id":request.build_absolute_uri(),
-                            # We probably need to create the group_key url, if not present as field in the model
-                             "group":{"@id":"https://ipmanager-test.lib.umd.edu/groups/henson20001",
-                                      "key":group,
-                                      "name":"Henson Collection"},
+                             "group":{"@id":request.build_absolute_uri(reverse("group_key", args=[group_key])),
+                                      "key":group_key,
+                                      "name":db_group.name},
                              "ip_address": ip,
-                             "contained": contained()})
-      else :
+                             "contained": contained(ip, group_key)})
+      except Group.DoesNotExist:
         return JsonResponse({"title": "Group not found", 
-                             "detail": "There is no group with the key 'nonexistent-group'.", 
+                             "detail": f"There is no group with the key {group_key}.", 
                              "status": 404}, status=404)
-      
-# When only IP is present, the we need to create url for each group and check for membership. 
-# For contained, we need to create a function to actually check for membership and return boolean value.
 
-def contained():
+def generateChecksArray(ip_address, all_groups, request):
+    result = []
+    for group in all_groups:
+        single_group = {}
```

Comment:
A better pattern here is to use a single dictionary literal to create the `single_group` dictionary, instead of defining it by each key separately:

```python
single_group = {
    "@id": request.build_absolute_uri(reverse("check")) + "?" + urlencode({"group":group.key, "ip":ip_address}),
    "group": build_group_metadata(request, group),
    "ip": ip_address,
    "contained": contained(ip_address, group.key),
}
```

This also uses the `build_group_metadata()` function described earlier, and fixes the key names (`"@group"` --> `"group"` and `"ip_address"` --> `"ip"`).